model/lifecycle.go

```diff
@@ -1267,13 +1267,16 @@ func createOneTask(id string, buildVarTask BuildVariantTaskUnit, project *Projec
 	isGithubCheck := false
 	if len(githubChecksAliases) > 0 {
 		var err error
-		isGithubCheck, err = githubChecksAliases.HasMatchingTask(project.FindProjectTask(buildVarTask.Name))
-		grip.Error(message.WrapError(err, message.Fields{
-			"message": "error checking if task matches aliases",
-			"version": v.Id,
-			"task":    buildVarTask.Name,
-			"variant": buildVarTask.Variant,
-		}))
+		name, tags, ok := project.GetTaskNameAndTags(buildVarTask)
```

Comment:
I'm confused by this addition. If this check is correct, that means that an entire task group was passed in to this function, and we would create a single task for the entire task group below wouldn't it?

Reply:
Good question; IncludeDependencies adds in tasks from a task group. https://github.com/evergreen-ci/evergreen/blob/365ddf0e50b93ebcc622781725e300742fd38b1a/model/dependencies.go#L59
(I verified this in the splunk logs also)

Reply:
That answers it for pairs, but what are we doing with the full task at the bottom of this function?

Reply:
I'm not sure what you're asking. All the expected tasks were created in my test (see https://evergreen-staging.corp.mongodb.com/version/6000ccf19dbe325bad786f43 for example), and the tasks all have the task group name / max hosts / order as part of their data.

Reply:
I'm just thinking that this is the wrong place to put this check because only actual tasks will ever get here. I don't see any expanding of task groups in this function, and whatever it returns looks like it will end up getting inserted into the db without many changes. So I think this code path won't ever hit https://github.com/evergreen-ci/evergreen/pull/4313/files#diff-321f6a0cef4adde81d7a3d369c12d6fe6d280d3d2a69ec8c8798b56e9056b119R1158